packages/grid/x-data-grid/src/models/gridFilterOperator.ts

```diff
@@ -3,6 +3,29 @@ import { GridFilterItem } from './gridFilterItem';
 import { GridCellParams } from './params/gridCellParams';
 import type { GridColDef } from './colDef/gridColDef';
 import type { GridValidRowModel } from './gridRows';
+import type { GridApiCommunity } from './api/gridApiCommunity';
+
+type ApplyFilterFnLegacy<R extends GridValidRowModel = any, V = any, F = V> = (
+  params: GridCellParams<R, V, F>,
+  apiRef: React.MutableRefObject<GridApiCommunity>,
```

Comment:
The v7->legacy is the only useful conversion, because legacy->v7 would force us to always use the legacy unefficient API. Creating the `cellParams` for the legacy API is the slow operation.

We can't provide a `noop` function because as you pointed out in another thread, some users might reuse our exported filter operators to create new operators.

Btw could you explain in more details how it break that demo? I'm not sure I understand. I'm adding a parameter to the function, so I would expect that to be a backwards-compatible change.

Reply:
> legacy->v7 would force us to always use the legacy unefficient API.

Right, I can see now how it doesn't make any sense.

> Btw could you explain in more details how it break that demo? I'm not sure I understand. I'm adding a parameter to the function, so I would expect that to be a backwards-compatible change.

(I've updated the demo link in the original comment because the link was broken).

Yes, adding a parameter to the function is not a breaking change as long as you don't wrap this function with the assumption that it only receives one argument, like in this demo:
https://github.com/mui/mui-x/blob/38733c626ea74819a6b0d90b7a2baa7ea8a024f1/docs/data/data-grid/filtering/CustomSelectionOperator.js#L57-L62

<img width="757" src="https://github.com/mui/mui-x/assets/13808724/3e819646-0d6a-4d4a-8cee-e5d7ca290773">

Maybe it's not a big deal, and we can update the demo assuming that it was buggy before because it didn't forward `...args`? 😅

Reply:
I see. All I need the `apiRef` for in there is the v7->legacy conversion. I see 2 options:
 - Update the demo and break this edge-case
 - Use a temporary global ref/variable to store the `apiRef`
Wdyt?

Reply:
> Use a temporary global ref/variable to store the apiRef

Is this going to work with multiple grids? We will need a way to identify which instance of apiRef to use, and I don't see how we can do that.

Reply:
It would just be a sync operation:

```ts
/* conversion */
export function convertFilterV7ToLegacy(fn: GetApplyFilterFnV7): GetApplyFilterFnLegacy {
  return (filterItem, column) => {
    // ...
    return (cellParams): boolean => {
      return filterFn(cellParams.value, cellParams.row, column, GLOBAL_API_REF);
    };
  };
}


/* filtering */
GLOBAL_API_REF = thisAPIRef;
doFiltering();
GLOBAL_API_REF = null;
```

Reply:
When is V7?

Reply:
> It would just be a sync operation:

That would work.
One thing that is still not solved is the code sample I posted in https://github.com/mui/mui-x/pull/9254#discussion_r1229849623
In order for it to work, one would have to override `getApplyFilterFnV7` instead of `getApplyFilterFn` in that snippet.
Do you see any other options?

Reply:
> The global ref wouldn't solve that? 

The global ref would make overriding `getApplyFilterFn` work, yes.
But wouldn't the grid ignore the overridden `getApplyFilterFn` function and use `getApplyFilterFnV7` instead?

Reply:
Done